File: 2274/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(maxConsecutive(2, 9, []int{4,6}))		// 3
	fmt.Println(maxConsecutive(6, 8, []int{7,6,8}))		// 0
	fmt.Println(maxConsecutive(3, 15, []int{7,9,13}))	// 4
	fmt.Println(maxConsecutive(28, 50, []int{35, 48}))	// 12
}

func maxConsecutive(bottom int, top int, special []int) int {
	ans := 0
	if len(special) == (top - bottom + 1) {
		return ans
	}

	sort.Ints(special)

	for i := 1; i < len(special); i++ {
		ans = max(ans, special[i] - special[i-1] - 1)
	}
	ans = max(ans, special[0] - bottom, top - special[len(special)-1])
	return ans
}

File: 2274/main.rs
struct Solution {}

impl Solution {
    pub fn max_consecutive(bottom: i32, top: i32, special: Vec<i32>) -> i32 {
        let mut special = special;
        special.push(bottom - 1);
        special.push(top + 1);
        special.sort();
        let mut ans = 0;
        for i in 0..special.len() - 1 {
            ans = ans.max(special[i+1] - special[i] - 1)
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::max_consecutive(3, 15, vec![7,9,13])); 
}